applications/jupyter-lab-hpc-native/tapisjob_app.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+echo "TACC: job ${SLURM_JOB_ID} execution at: $(date)"
+
+# TAP Port
+LOCAL_PORT=5902
+
+module load python3/3.9.2
+module load jupyterlab
+
+# Ensure TAP functions are available
+TAP_FUNCTIONS="/share/doc/slurm/tap_functions"
+if [ -f ${TAP_FUNCTIONS} ]; then
+    . ${TAP_FUNCTIONS}
+else
+    echo "TACC: ERROR - could not find TAP functions file: ${TAP_FUNCTIONS}"
+    exit 1
+fi
+
+# Set LOGIN_PORT using tap_get_port and verify it succeeds
+LOGIN_PORT=$(tap_get_port)
+if [ -z "${LOGIN_PORT}" ]; then
+    echo "TACC: ERROR - could not retrieve login port."
+    exit 1
+fi
+
+# Retrieve node domain for Jupyter URL
+NODE_HOSTNAME_DOMAIN=$(hostname -d)
+
+# Generate or retrieve TAP_TOKEN
+TAP_CERTFILE="${HOME}/.tap/.${SLURM_JOB_ID}"
+TAP_TOKEN=$(tap_get_token)
+if [ -z "${TAP_TOKEN}" ]; then
+    echo "TACC: ERROR - could not generate token for notebook"
+    exit 1
+fi
+echo "TACC: using token ${TAP_TOKEN}"
+
+# Define Jupyter Arguments
+JUPYTER_ARGS="--port=${LOCAL_PORT} --certfile=$(cat ${TAP_CERTFILE}) --ServerApp.allow_remote_access=True --ServerApp.token=${TAP_TOKEN}"
```

Comment:
Any reason this is defining args here rather than using the config file?

Reply:
e.g. https://github.com/TACC/WMA-Tapis-Templates/blob/4eca35f4722c59e6e620ebbfa76ac117c7cc698d/applications/jupyter-lab-hpc/run.sh#L31-L32